kcqrs-core/src/main/kotlin/com/clouway/kcqrs/core/AuthoredEvent.kt

```diff
@@ -0,0 +1,13 @@
+package com.clouway.kcqrs.core
+
+/**
+ * AuthoredEvent will be used when you want to have the Identity of
+ * the event provided, e.g. you want the id of the author of the event
+ * and the time of the event.
+ *
+ * @author Vasil Mitov <vasil.mitov@clouway.com>
+ */
+
+abstract class AuthoredEvent private constructor(var identity: Identity?) : Event {
```

Comment:
Doesn't look reasonable to me as the EventStore is already returning the EventPayload in the body of the Aggregate.

```kotlin
data class EventPayload(val kind: String, val timestamp: Long, val identityId: String, val data: Binary)
```

e.g this information could be provided to Aggregate and to be used by it or to be provided separately by a context but it will enforce an API change. 


Optional A:
```kotlin
val history = aggregate.auditHistory() 
// identityId, timestamp, eventType 
// 123, 123123123123, 'InvoiceCreatedEvent'
// 123, 123123123124, 'InvoiceAddressChangedEvent'
```

or 

```kotlin
val aggregateContext = aggregateRepository.getById("Test", Invoice::class.java) // returns AggregateContext<Invoice>

aggregateContext.value()
aggregateContext.auditLog()
// identityId, timestamp, eventType 
// 123, 123123123123, 'InvoiceCreatedEvent'
// 123, 123123123124, 'InvoiceAddressChangedEvent'
```

Reply:
It could be added, but the idea is to have the information upon event handling.

Reply:
Yes, and the API of the EventHandler may be changed to accept the EventContext in this situation:

```kotlin
fun <T>  handle(context: EventContext, event: T) 
```

whether the aggregate context will hold the identity and the time of the event.

Reply:
or just:

```kotlin
fun handle(context: EventContext)
```
and the event to be contained in the event context.

Reply:
Another 2 cents on this flow: 

As the basic flow is:
client app calls event store and event-store publishes events to the messaging bus 

e.g the EventPublisher need to provide this information to the message bus and the message bus should be able to send it as metadata to the target callers.

Reply:
It could be done like that. It would force a change across the existing handlers for the Events, but the code would look better. Maybe i can think of some sort of migration bridge for now to work with both event and context.

Reply:
just : 
```kotlin
fun handle(context: EventContext)
```
will not be enough, maybe 
```
fun handle(context: EventContext<Event>)
```
or 
```kotlin
fun handle(event:Event, context: EventContext)
```